kong/pdk/request.lua

```diff
@@ -235,6 +236,41 @@ local function new(self)
   end
 
 
+  ---
+  -- Returns the path component of the request's URL, but also considers
+  -- `X-Forwarded-Prefix` if it comes from a trusted source. The value
+  -- is returned as a Lua string.
+  --
+  -- Whether this function considers `X-Forwarded-Prefix` or not depends on
+  -- several Kong configuration parameters:
+  --
+  -- * [trusted\_ips](https://getkong.org/docs/latest/configuration/#trusted_ips)
+  -- * [real\_ip\_header](https://getkong.org/docs/latest/configuration/#real_ip_header)
+  -- * [real\_ip\_recursive](https://getkong.org/docs/latest/configuration/#real_ip_recursive)
+  --
+  -- **Note**: we do not currently do any normalization on the request
+  --           path except return `"/"` on empty path.
```

Comment:
This seems false or incomplete, there is some normalization going on on the last line of this function.

Reply:
Thus it says:
> except return `"/"` on empty path.

Am I missing something?

Reply:
I agree with @bungle that the note seems in accordance with the code. But I do not understand why this normalization is necessary - but there is probably a reason for it. ;-) What would be wrong with returning just path (even emtpy string), in case the X-Forwarded-Prefix header is not set by a trusted client?

Reply:
that is the problem of these proprietary headers. There is no documentation. So I checked what others do and came across this: https://github.com/qaware/x-forwarded-filter

e.g. it normalizes request url like:
`/a/b/../` to `/a`. I considered it but then just went with "" -> "/". We can add more later if needed, but we need to be careful to not mess up with semantics.